lib/active_model/serializer/adapter/attributes.rb

```diff
@@ -2,42 +2,58 @@ module ActiveModel
   class Serializer
     module Adapter
       class Attributes < Base
+        attr_accessor :_nested_associations
+
         def serializable_hash(options = nil)
           options ||= {}
+
+          self._nested_associations = ActiveModel::Serializer::Utils
+            .include_args_to_hash(options[:include])
+
           if serializer.respond_to?(:each)
-            result = serializer.map { |s| Attributes.new(s).serializable_hash(options) }
+            serializable_hash_for_collection(options)
           else
-            hash = {}
+            serializable_hash_for_single_resource(options)
+          end
+        end
+
+        def serializable_hash_for_collection(options = nil)
+          serializer.map { |s| Attributes.new(s).serializable_hash(options) }
+        end
+
+        def serializable_hash_for_single_resource(options = {})
+          resource = resource_object_for(serializer, options)
+          add_resource_relationships(resource, serializer)
```

Comment:
Could we not have `relationships = resource_relationships_for(serializer); resource.merge!(relationships)`? It's important that we avoid side-effects as much as possible.